sys/net/gnrc/network_layer/ipv6/nc/gnrc_ipv6_nc.c

```diff
@@ -163,6 +163,14 @@ void gnrc_ipv6_nc_remove(kernel_pid_t iface, const ipv6_addr_t *ipv6_addr)
 #endif
 #ifdef MODULE_GNRC_SIXLOWPAN_ND_ROUTER
         xtimer_remove(&entry->type_timeout);
+
+        gnrc_ipv6_netif_t *if_entry = gnrc_ipv6_netif_get(iface);
+
+        if ((if_entry != NULL) &&
+              (if_entry->rtr_adv_msg.content.ptr == (char *) entry)) {
+            // cancel timer set by gnrc_ndp_rtr_sol_handle
+            xtimer_remove(&if_entry->rtr_adv_timer);
```

Comment:
What if this timer was set by a router solicitation of another neighbor. I think the better solution would be to check if the neighbor cache entry is still set in the handler function of this timer.